intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduFullPluginVersion.kt

```diff
@@ -0,0 +1,72 @@
+package com.jetbrains.plugin.structure.edu
+
+import com.jetbrains.plugin.structure.base.plugin.PluginProblem
+import com.jetbrains.plugin.structure.base.problems.PropertyNotSpecified
+import com.jetbrains.plugin.structure.base.utils.CompatibilityUtils
+import com.jetbrains.plugin.structure.edu.problems.InvalidVersionError
+
+
+data class EduFullPluginVersion(
+  val pluginVersion: EduPluginVersion,
+  val ideVersion: String,
+  val buildNumber: String
+): Comparable<EduFullPluginVersion> {
+  companion object {
+    fun fromString(version: String?): EduFullPluginVersion {
+      if (version.isNullOrEmpty()) fail(version)
+      val parts = version.split("-")
+      if (parts.size != 3 || parts.any { it.isEmpty() }) fail(version)
+      return EduFullPluginVersion(EduPluginVersion.fromString(parts[0]), parts[1], parts[2])
+    }
+
+    fun createIfValid(version: String?, problems: MutableList<PluginProblem>): EduFullPluginVersion? {
+      if (version.isNullOrBlank()) {
+        problems.add(PropertyNotSpecified(VERSION))
+        return null
+      }
+
+      return try {
```

Comment:
runCathing

Reply:
I think that it's awful, may be I miss something:

```
      runCatching {
        return fromString(version)
      }.onFailure {
        problems.add(InvalidVersionError(version))
      }
      
      return null
```

Reply:
```kotlin
  val orElse = runCatching { fromString(version) }.getOrElse {
    problems.add(PropertyNotSpecified(VERSION))
    null
  } ?: return null
```
=)

but the same ((